Simplenote/PDFLinkActivityItemProvider.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+public class PDFLinkActivityItemProvider : UIActivityItemProvider {
+    public let targetURL: URL
+    public let content: String
+    private lazy var documentURL = shareToPDF()
+    
+    public init(content: String, filename: String) {
+        self.targetURL = FileManager.default.temporaryDirectory.appendingPathComponent(filename)
+        self.content = content
+        
+        super.init(placeholderItem: targetURL)
+    }
+    
+    public override var item: Any {
+        documentURL ?? content
+    }
+    
+    private func shareToPDF() -> URL? {
+        guard activityType?.rawValue == Consts.remoteOpenInAppActivityType else {
+            return nil
+        }
+        let data = SimplenotePDFExporter.exportStringToPDFData(content)
+        return FileManager.writeDataToDocuments(data: data, to: targetURL)
+    }
```

Comment:
How about if instead of coupling the output of this API to the `activityType` instead we implement this API?
(And drop the `item: Any` getter)

```
    public override func activityViewController(_ activityViewController: UIActivityViewController, itemForActivityType activityType: UIActivity.ActivityType?) -> Any? {
        guard activityType?.rawValue == Consts.remoteOpenInAppActivityType else {
            return nil
        }

        return documentURL ?? content
    }
```